src/device_stm32h7.c

```diff
@@ -13,12 +13,18 @@
 #define FLASH_CCR 0x14
 #define FLASH_OPTSR_CUR 0x1c
 #define FLASH_OPTSR_PRG 0x20
+#define SCB (0xE000E000UL +  0x0D00UL)
+#define SCB_VTOR  (SCB + 0x008)
 
 void *mg_flash_start(void) {
   return (void *) 0x08000000;
 }
 size_t mg_flash_size(void) {
-  return 2 * 1024 * 1024;  // 2Mb
+#ifdef MG_DEVICE_SINGLE_BANK
+  return 1024 * 1024;  // 1Mb
+#else
+  return 2 * 1024 * 1024; // 2Mb;
+#endif
```

Comment:
Instead, read flash size from a device.
For example, [RM0433](https://www.st.com/resource/en/reference_manual/rm0433-stm32h742-stm32h743753-and-stm32h750-value-line-advanced-armbased-32bit-mcus-stmicroelectronics.pdf) for H743 in section 61.1 describes the Unique device ID register. It is a unique 96-bit STM32 identifier.
What follows in section 61.2 is the flash size register: address 0x1FF1E880 which contains the size in kB. Let's read it instead of hardcoding the values.